ale_linters/erlang/erlc.vim

```diff
@@ -3,7 +3,8 @@
 let g:ale_erlang_erlc_options = get(g:, 'ale_erlang_erlc_options', '')
 
 function! ale_linters#erlang#erlc#GetCommand(buffer) abort
-    return 'erlc ' . g:ale_erlang_erlc_options . ' %t'
+    let l:temp_dir = has('win32') ? $TMP : $TMPDIR
```

Comment:
Are we walking about `CreateTemporaryFileForJob`? That could be useful here, although it's not an exported function (yet). I _do_ see the `ManageFile` and `ManageDirectory` functions too.